### PR TITLE
Fix ARGF.read interprets the certname as a filename and attempts to read it

### PR DESCRIPTION
**Pull request by rjpearce**
This [commit](https://github.com/danieldreier/autosign/commit/faf96a198aeb643a5b74cdf570498fe33f22f4b8) has broken autosign_validate.

tldr: ARGF.read interprets the certname as a filename and attempts to read it

Example:
cat test-174.bah.csr | /opt/puppetlabs/puppet/bin/autosign-validator test174.bah
Traceback (most recent call last):
        3: from /opt/puppetlabs/puppet/bin/autosign-validator:23:in `<main>'
        2: from /opt/puppetlabs/puppet/bin/autosign-validator:23:in `load'
        1: from /opt/puppetlabs/puppet/lib/ruby/gems/2.5.0/gems/autosign-0.1.3 on/bin/autosign-validator:6:in `<top (required)>'
/opt/puppetlabs/puppet/lib/ruby/gems/2.5.0/gems/autosign-0.1.3/bin/autosign-validator:6:in `read': No such file or directory @ rb_sysopen - test174.bah (Errno::ENOENT)

Detailed Explanation:
According to ARGF(https://ruby-doc.org/core-2.5.0/ARGF.html)
"The arguments passed to your script are stored in the ARGV Array, one argument per element. ARGF assumes that any arguments that aren't filenames have been removed from ARGV"

The first and only argument here is an certname
ARGF.read interprets the certname as a filename and attempts to read it but fails as the file does not exist.

The fix is to use ARGV.shift to pull in the certname and then call ARGF.read.

### CODE REVIEW

File: bin/autosign-validator [hunk 1]
```diff
@@ -2,9 +2,6 @@
 require 'autosign'
 require 'logging'
 
-### Read all the arugments https://tickets.puppetlabs.com/browse/SERVER-1116
-argf_read = ARGF.read
```
Comment (danielparks) [inline bin/autosign-validator]: It would be cleanest to just change this to replace with with the `raw_csr = $stdin.read` lines from below. No need to futz with `ARGF`.

None of the other changes would be necessary, I think.

Comment (rjpearce) [reply bin/autosign-validator]: @danielparks I have just added another commit to switch it back to $stdin.read. 

Do you also have access to review/merge this: https://github.com/danieldreier/puppet-autosign/pull/29

**Comment by rjpearce** [comment]
Do you also have access to review/merge this: danieldreier/puppet-autosign#29

File: bin/autosign-validator [hunk 2]
```diff
@@ -2,6 +2,8 @@
 require 'autosign'
 require 'logging'
 
+raw_csr = $stdin.read
```
Comment (danielparks) [inline bin/autosign-validator]: Could you add a comment (the old one is  fine) explaining why it needs to read all the input?

**Comment by danielparks** [comment]
It’s up! https://rubygems.org/gems/autosign

**Comment by rjpearce** [comment]
@Danielparks thanks. Do you have access to pull this one on the forge module https://github.com/danieldreier/puppet-autosign/pull/29?
